src/main/scripts/install.sh

```diff
@@ -0,0 +1,215 @@
+    fi
+
+    echo "cluster \"${clusterName}\" is successfully created!"
+}
+
+create_custom_profile() {
+    profileName=$1
+    hostName=$2
+    nodeName=$3
+    dmgrHostName=$4
+    dmgrPort=$5
+    dmgrAdminUserName=$6
+    dmgrAdminPassword=$7
+    
+    curl $dmgrHostName:$dmgrPort --output - >/dev/null 2>&1
+    while [ $? -ne 56 ]
+    do
+        sleep 5
+        echo "dmgr is not ready"
+        curl $dmgrHostName:$dmgrPort --output - >/dev/null 2>&1
+    done
+    sleep 60
+    echo "dmgr is ready to add nodes"
+
+    output=$(${WAS_ND_INSTALL_DIRECTORY}/bin/manageprofiles.sh -create -profileName $profileName -hostName $hostName -nodeName $nodeName \
+        -profilePath ${WAS_ND_INSTALL_DIRECTORY}/profiles/$profileName -templatePath ${WAS_ND_INSTALL_DIRECTORY}/profileTemplates/managed \
+        -dmgrHost $dmgrHostName -dmgrPort $dmgrPort -dmgrAdminUserName $dmgrAdminUserName -dmgrAdminPassword $dmgrAdminPassword 2>&1)
+    while echo $output | grep -qv "SUCCESS"
+    do
+        sleep 10
+        echo "adding node failed, retry it later..."
+        rm -rf ${WAS_ND_INSTALL_DIRECTORY}/profiles/$profileName
+        output=$(${WAS_ND_INSTALL_DIRECTORY}/bin/manageprofiles.sh -create -profileName $profileName -hostName $hostName -nodeName $nodeName \
+            -profilePath ${WAS_ND_INSTALL_DIRECTORY}/profiles/$profileName -templatePath ${WAS_ND_INSTALL_DIRECTORY}/profileTemplates/managed \
+            -dmgrHost $dmgrHostName -dmgrPort $dmgrPort -dmgrAdminUserName $dmgrAdminUserName -dmgrAdminPassword $dmgrAdminPassword 2>&1)
+    done
+    echo $output
+}
+
+while getopts "m:c:f:h:r:x:" opt; do
+    case $opt in
+        m)
+            adminUserName=$OPTARG #User id for admimistrating WebSphere Admin Console
+        ;;
+        c)
+            adminPassword=$OPTARG #Password for administrating WebSphere Admin Console
+        ;;
+        f)
+            dmgr=$OPTARG #Flag indicating whether to install deployment manager
+        ;;
+        h)
+            dmgrHostName=$OPTARG #Host name of deployment manager server
+        ;;
+        r)
+            members=$OPTARG #Number of cluster members
+        ;;
+        x)
+            dynamic=$OPTARG #Flag indicating whether to create a dynamic cluster or not
+        ;;
+    esac
+done
+
+# Check whether the user is entitled or not
+while [ ! -f "/var/log/cloud-init-was.log" ]
+do
+    sleep 5
+done
+
+isDone=false
+while [ $isDone = false ]
+do
+    result=`(tail -n1) </var/log/cloud-init-was.log`
+    if [[ $result = Unentitled ]] || [[ $result = Entitled ]]; then
+        isDone=true
+    else
+        sleep 5
+    fi
+done
+echo "The input IBMid account is ${result}."
+
+# Remove cloud-init artifacts and logs
+cloud-init clean --logs
+
+# Terminate the process for the un-entitled user
+if [ ${result} = Unentitled ]; then
+    exit 1
+fi
+
+# Update applications installed on the system
+yum update -y
+
+# Turn off firewall
```

Comment:
Stopping and disabling the firewall, is a security concern. Instead we should open the ports which are needed.

Reply:
I agree. The reason why turning off the firewall is to make the federation of custom nodes work. @git4rk Pls tell me exact ports I should open, so that the following command (there could be more, depending on the testing result after applying changes) can be successfully executed:
* `${WAS_ND_INSTALL_DIRECTORY}/bin/manageprofiles.sh -create -profileName $profileName -hostName $hostName -nodeName $nodeName \
        -profilePath ${WAS_ND_INSTALL_DIRECTORY}/profiles/$profileName -templatePath ${WAS_ND_INSTALL_DIRECTORY}/profileTemplates/managed \
        -dmgrHost $dmgrHostName -dmgrPort $dmgrPort -dmgrAdminUserName $dmgrAdminUserName -dmgrAdminPassword $dmgrAdminPassword`
 * ``

Reply:
@git4rk I found an [relative article](http://www.freekb.net/Article?id=1544) talking about the ports need to be opened for federating nodes into the deployment manager, which seems worthy to have a try. I will test it today and let you know the result.

Reply:
The above reference works based on my testing, code changes is pushed with bcb6e97.